test/functional/tool_signet_miner.py

```diff
@@ -0,0 +1,72 @@
+#!/usr/bin/env python3
+# Copyright (c) 2022 The Bitcoin Core developers
+# Distributed under the MIT software license, see the accompanying
+# file COPYING or http://www.opensource.org/licenses/mit-license.php.
+"""Test signet miner tool"""
+
+import os.path
+import subprocess
+import sys
+import time
+
+from test_framework.key import ECKey
+from test_framework.script_util import keys_to_multisig_script
+from test_framework.test_framework import (
+    BitcoinTestFramework,
+    SkipTest,
+)
+from test_framework.util import assert_equal
+from test_framework.wallet_util import bytes_to_wif
+
+
+CHALLENGE_PRIVATE_KEY = (42).to_bytes(32, 'big')
+
+
+class SignetMinerTest(BitcoinTestFramework):
+    def set_test_params(self):
+        self.chain = "signet"
+        self.setup_clean_chain = True
+        self.num_nodes = 1
+
+        # generate and specify signet challenge (simple 1-of-1 multisig script)
+        privkey = ECKey()
+        privkey.set(CHALLENGE_PRIVATE_KEY, True)
+        pubkey = privkey.get_pubkey().get_bytes()
+        challenge = keys_to_multisig_script([pubkey])
+        self.extra_args = [[f'-signetchallenge={challenge.hex()}']]
+
+    def skip_test_if_missing_module(self):
+        self.skip_if_no_cli()
+        self.skip_if_no_wallet()
+        self.skip_if_no_bdb()
+
+    def run_test(self):
+        node = self.nodes[0]
+        # import private key needed for signing block
+        node.importprivkey(bytes_to_wif(CHALLENGE_PRIVATE_KEY))
+
+        # determine paths to tools
+        base_dir = self.config["environment"]["SRCDIR"]
+        signet_miner_path = os.path.join(base_dir, "contrib", "signet", "miner")
+        # TODO: this is a dirty hack; the path to bitcoin-util should ideally
```

Comment:
I don't think it's good to leave it like this, if we use `bitcoin-util` in the functional tests we'd at least need to

- Add a `self.options.bitcoinutil` in `BitcoinTestFramework.setup`, analogous to the other two, overridable with `BITCOINUTIL`

- we also need an entry in `config.ini.in` for `BUILD_BITCOIN_UTIL` → an `is_bitcoin_util_compiled()` → and an `skip_if_no_bitcoin_util`

Reply:
Thanks for the guide, will add a commit with these preliminaries within the next days.